dev/source/docs/setting-up-sitl-on-macosx.rst

```diff
@@ -0,0 +1,137 @@
+.. _setting-up-sitl-on-macosx:
+
+===========================
+Setting up SITL on Mac OS X
+===========================
+
+This page describes how to setup the :ref:`SITL (Software In The Loop) <sitl-simulator-software-in-the-loop>` on Mac OS X. The specific commands were tested on macOS Catalina 10.15.7 and macOS Big Sur 11.1.
+
+Overview
+========
+
+The SITL simulator allows you to run Plane, Copter or Rover without any
+hardware. It is a build of the autopilot code using an ordinary C++
+compiler, giving you a native executable that allows you to test the
+behaviour of the code without hardware.
+
+SITL runs natively on Linux, Windows and Mac OSX.
+
+.. image:: ../images/SITL_macosx.png
+    :target: ../_images/SITL_macosx.png
+
+
+Install steps
+=============
+
+This guide assumes that a build environment is already setup for Ardupilot. If that is not the case please follow the available guides to setup the build environment for Mac OSX. The following step provide a condensed versions of the steps to take.
+
+
+Prerequisites
+-------------
+
+The following tools need to be installed to complete the installation. Between brackets the version used by the author at time of writing.
+
+ * Xcode (12.4)
+ * Xcode commandline tools (12.4)
+ * Homebrew (3.0.4-52-gbc37074)
+
+Preparing Python
+----------------
+
+Install the required python 3 version using homebrew and activate it. If you are familiar with pyenv or virtualenv it might be better to use one of those to setup the environment. This limits potential problems with other tools that require specific versions of python. How to do that is beyond the scope of this document.
+
+The following steps install python version 3.9.x for system wide usage.
+
+::
+
+	brew install python@3.9
+  export PATH="$(brew --prefix)/opt/python/libexec/bin":$PATH
+	
+	pip install wheel
+	pip install future pymavlink mavproxy
```

Comment:
Using Python 3.10 from Python.org for Mac M1 ARM version, this will fail with the following errors:
wx/svg/_nanosvg.c:20946:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                      ^
    wx/svg/_nanosvg.c:20946:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    wx/svg/_nanosvg.c:20951:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                      ^
    wx/svg/_nanosvg.c:20951:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    wx/svg/_nanosvg.c:21035:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, NULL);
                      ^
    wx/svg/_nanosvg.c:21035:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, NULL);

Reply:
I'm not totally sure what these lines do, but it seems that there is some code in _nanosvg.c that needs to be fixed.